src/auth/auth-get-domain.ts

```diff
@@ -0,0 +1,29 @@
+
+const thirdPartyDomainSTSUrlMappings: { [domain: string]: string; } = {
+  'bbk12.com': 'https://sts-sso.bbk12.com'
+};
+const defaultSTSUrl = 'https://s21aidntoken00blkbapp01.nxt.blackbaud.com';
+
+export class BBAuthDomain {
+
+  public static CURRENT_DOMAIN = window.location.hostname;
+
+  public static getRegisteredDomain(): string {
+    let domain;
+
+    Object.keys(thirdPartyDomainSTSUrlMappings).forEach((d) => {
```

Comment:
I would use `Object.keys().find()` to simplify this code. Doing so will break out of the loop when the first matching item is found, rather than continue to loop as the current code would.

```
return Object.keys(thirdPartyDomainSTSUrlMappings).find((d) => {
  const domainEndCompare = `.${d}`;

  return this.CURRENT_DOMAIN === d || this.CURRENT_DOMAIN.substr(-domainEndCompare.length) === domainEndCompare;
});
```